python/pyspark/ml/image.py

```diff
@@ -0,0 +1,135 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark import SparkContext
+from pyspark.sql.types import Row, _create_row, _parse_datatype_json_string
+from pyspark.sql import DataFrame, SparkSession
+import numpy as np
+
+undefinedImageType = "Undefined"
+
+imageFields = ["origin", "height", "width", "nChannels", "mode", "data"]
+
+
+class _ImageSchema(object):
+    """
+    Returns the image schema.
+
+    :rtype StructType: a DataFrame with a single column of images
+           named "image" (nullable)
+
+    .. versionadded:: 2.3.0
+    """
+    @property
+    def imageSchema(self):
+        ctx = SparkContext._active_spark_context
+        jschema = ctx._jvm.org.apache.spark.ml.image.ImageSchema.imageSchema()
+        return _parse_datatype_json_string(jschema.json())
+
+    """
+    Returns the OpenCV type mapping supported
+
+    :rtype dict: The OpenCV type mapping supported
+
+    .. versionadded:: 2.3.0
+    """
+    @property
+    def ocvTypes(self):
+        ctx = SparkContext._active_spark_context
+        return ctx._jvm.org.apache.spark.ml.image.ImageSchema._ocvTypes()
+
+ImageSchema = _ImageSchema()
```

Comment:
Ah .. thing is, docstring .. let me try to think more tonight (KTS) and probably make a cleanup PR to your branch, @imatiach-msft so that that can be pused to this PR, or leave another comment here.

Reply:
sure, you are welcome to add a cleanup PR to my branch, or you can just comment on what I should update, either way works